src/main/scala/mesosphere/marathon/core/election/ElectionService.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.marathon.core.election
+
+import scala.concurrent.Future
+
+trait LeadershipAbdication {
+  def abdicateLeadership(error: Boolean = false): Unit
+}
+
+trait ElectionService extends LeadershipAbdication {
+  def isLeader: Boolean
+  def leader: Option[String]
+  def offerLeadership(): Unit
+}
+
+trait ElectionDelegate {
```

Comment:
Just looking into trait definitions, it is not clear what an ElectionDelegate is.
1) Perhaps we need a better name?
2) Please add some class level documentation.